widget/list.go

```diff
@@ -409,12 +459,26 @@ func (l *listLayout) Layout([]fyne.CanvasObject, fyne.Size) {
 }
 
 func (l *listLayout) MinSize([]fyne.CanvasObject) fyne.Size {
-	if f := l.list.Length; f != nil {
-		separatorThickness := theme.Padding()
-		return fyne.NewSize(l.list.itemMin.Width,
-			(l.list.itemMin.Height+separatorThickness)*float32(f())-separatorThickness)
+	items := 0
+	if f := l.list.Length; f == nil {
+		return fyne.NewSize(0, 0)
+	} else {
+		items = f()
 	}
-	return fyne.NewSize(0, 0)
+
+	height := float32(0)
+	templateHeight := l.list.itemMin.Height
+	for item := 0; item < items; item++ {
```

Comment:
This is, I fear, going against the current efficiency of a list and might trigger performance regression. It would be safer to iterate over the itemHeights content and then add the templateHeight * unknown size item. This would make performance for current user of the list not change.

Reply:
We are just iterating an index, I think it would require benchmarks to show at what number of overrides this is slower than all the map accesses and calculating the gaps and doing the maths at the end.
As above it is the same code we have been using in Table for some time. Does it need fixed there as well, or can we return to it later as we optimise?